gui/launcher.h

```diff
@@ -124,11 +155,77 @@ class LauncherDialog : public Dialog {
 	 *
 	 * @target	name of target to select
 	 */
-	void selectTarget(const String &target);
+	virtual void selectTarget(const String &target) = 0;
+	virtual int getSelected() = 0;
 private:
 	bool checkModifier(int modifier);
 };
 
+// Disable the grid for platforms that disable fancy themes
+class LauncherChooser {
+protected:
+	LauncherDialog *_impl;
+
+public:
+	LauncherChooser();
+	~LauncherChooser();
+
+	int runModal();
+	void selectLauncher();
+};
+
+class LauncherSimple : public LauncherDialog {
+public:
+	LauncherSimple(const U32String &title);
+
+	void handleCommand(CommandSender *sender, uint32 cmd, uint32 data) override;
+	void handleKeyDown(Common::KeyState state) override;
+
+#ifndef DISABLE_LAUNCHERDISPLAY_GRID
+	LauncherDisplayType getType() const override { return kLauncherDisplayList; }
+#endif // !DISABLE_LAUNCHERDISPLAY_GRID
+
+protected:
+	void updateListing() override;
+	void updateButtons() override;
+	void selectTarget(const String &target) override;
+	int getSelected() override;
+	void build() override;
+private:
+	ListWidget 		*_list;
+	ButtonWidget	*_addButton;
+	Widget			*_startButton;
+	ButtonWidget	*_loadButton;
+	Widget			*_editButton;
+	Widget			*_removeButton;
+};
+
+#ifndef DISABLE_LAUNCHERDISPLAY_GRID
```

Comment:
This, and the relevant bit in the LauncherChooser, is probably the only parts of this file that need to be guarded.

Reply:
I have removed unnecessary directives in 215ddc4.